docs: consolidated tech docs (#1297)
aave/aave-v4: docs/Glossary.md

Reviewer: I asked cc @EthWarrior his thoughts here too

Reviewer: He says Canonical is good, I like it too

# Glossary

Canonical terminology for Aave V4. Terms appear in alphabetical order.

---

**Base Drawn Rate:** Per-asset interest rate set at the Hub level, applied to all drawn debt for that asset. Determined by the Hub's interest rate strategy, based on current utilization. The symbol $R_{sbase,i}$ refers to the base drawn rate for asset $i$.

**Collateral Factor:** Maximum percentage of collateral value that can be borrowed against a collateral asset.

> **Collateral Risk:** Additional risk parameter per asset, managed by the Governor. Introduces a charge on borrowers above the base drawn interest rate.

Author: ```suggestion
**Collateral Risk:** Additional risk parameter per asset, managed by the Governor. Introduces a charge on borrowers above the base drawn rate.
```

just for consistency with term above

Author: later when we have risk stewards able to update CR, how to classify that. Not really only managed by the governor only, thoughts? am i reading too much into this

Reviewer 2: I see your point, but not sure since that's our implementation and not a codebase default

Author: ok makes sense, then we're just considering them governors in that case. np

> **Deficit:** Protocol accounting state in which a borrower has outstanding debt but no remaining collateral after a liquidation. Deficits are tracked per asset at the Hub level and can be eliminated by authorized active Spokes using their supplied shares.

Author: hm perhaps this should mention deficits as per spoke as well, and eliminating is by spokes on behalf of specific other spokes' reported deficit, wdyt @CanonicalJP

> **Draw Cap:** Maximum amount a Spoke can draw from the Hub.

Author: ```suggestion
**Draw Cap:** Maximum amount a spoke can draw from a hub.
```

Reviewer 2: capitalization is correct in this context, will make the change of `the` for `a`

> **DrawnDebt:** Base debt of a position, accruing interest at the base borrow rate.

Author: ```suggestion
**DrawnDebt:** Base debt of a position, accruing interest at the base drawn rate.
```

**Dynamic Risk Configuration:** Combination of Risk Configuration parameters that allows different users to be subscribed to different configurations.

**Governor:** Entity authorized to manage Spokes and Hubs, set risk parameters, configure caps, and execute emergency controls. In the Aave V4 deployment, this is the Aave DAO.

> **Health Factor (HF):** Ratio of a user's weighted collateral value (adjusted by Collateral Factor) to total outstanding debt. When HF drops below `HEALTH_FACTOR_LIQUIDATION_THRESHOLD`, the position becomes eligible for liquidation.

Author: ```suggestion
**Health Factor (HF):** Ratio of a user's weighted collateral value (adjusted by Collateral Factor) to total outstanding debt value. When HF drops below `HEALTH_FACTOR_LIQUIDATION_THRESHOLD`, the position becomes eligible for liquidation.
```

> **Hub:** Central store where all liquidity is held, to be drawn and restored by attached Spokes. No collateralization is enforced on the Hub.

Author: and added/removed?

Reviewer 2: good point, I'm rephrasing to avoid listing all actions

> **Isolation Mode:** Configuration pattern that deploys a dedicated Spoke and Liquidity Hub pair for a new or experimental asset with contained risk. This asset is the only allowed collateral asset. Borrowable assets are restricted by the `borrowable` flag on each reserve; maximum borrow exposure is controlled per asset through `SpokeData.drawCap` at the Hub. Users retain unrestricted positions in other Spokes simultaneously.

Author: can we settle on Liquidity Hub or just Hub? should the above be Liquidity Hub then, or like "Hub or Liquidity Hub"

Author: ```suggestion
**Isolation Mode:** Configuration pattern that deploys a dedicated Spoke and Liquidity Hub pair for a new or experimental asset with contained risk. This asset is the only allowed collateral asset. Borrowable assets are restricted by the `borrowable` flag on each reserve; maximum borrow exposure is controlled per asset through `SpokeData.drawCap` at the Hub. Users can retain positions in other Spokes simultaneously using the same wallet.
```

smt like this?

Reviewer 2: Any occurrence of "Liquidity Hub" is a miss in the Style Guide implementation
